gateway/crates/gateway-binary/tests/mocks/uplink.rs

```diff
@@ -0,0 +1,33 @@
+use std::ops::Deref;
+use std::str::FromStr;
+
+use ulid::Ulid;
+
+use federated_server::UplinkResponse;
+
+#[derive(serde::Serialize, serde::Deserialize)]
+pub(crate) struct UplinkResponseMock(UplinkResponse);
```

Comment:
Not sure why you did a `UplinkResponseMock`, a single function like `fn create_mock_uplink_response(sdl: String) -> UplinkResponse` seems to do the job in a simpler way.

Reply:
yeah contemplated it but ended up with this to add functionality to the mock - like `as_json`. Might be useful in the future or perhaps not, if it turns out its not I'll replace it with a simpler function.

Reply:
not sure to see how it brings value compared to just `serde_json::to_string` but 🤷